src/handlers/note.rs

```diff
@@ -0,0 +1,107 @@
+//! Allow users to add summary comments in Issues & Pull Requests.
+//!
+//! Users can make a new summary entry by commenting the following:
+//!
+//! ```md
+//! @rustbot note summary-title
+//!
+//! ...details details details...
```

Comment:
nit: let's drop this from the description of notes, since the implementation doesn't require details afterwards and we probably don't necessarily want to require that either. (If we ever start trying to capture those automatically it'll probably be by scraping the entire comment text, I suspect).